lib/oauth2.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
```

Comment:
Because we only use this for mail loading purposes, let's rename this file and refactor it so that it just returns the IMAP "mail" object, e.g. the return value of the various classes' "get_all_mail_folder()" methods. That way, we can just have those classes call `${class_name}.load_mail()` or whatever else we want to call the class. 

Then, we should refactor the methods so that they can use either way of authentication -- either the old app-password method or the new oauth mechanism. That way, this won't break everyone using the old method

Reply:
Probably also a second method to get the email sender, used in lib/email_sender.py